tests/unit/amplitude/test_parity_prefactor.py

```diff
@@ -97,40 +88,11 @@ def test_parity_prefactor(
         filename=f'amplitude_model_prefactor_{"-".join(test_input.intermediate_states)}.xml',
     )
 
-    prefactor1 = extract_prefactor(amplitude_model, related_component_names[0])
-    prefactor2 = extract_prefactor(amplitude_model, related_component_names[1])
-
-    assert prefactor1 == relative_parity_prefactor * prefactor2
-
-
-def extract_prefactor(amplitude_dict, coefficient_amplitude_name):
-    dict_element_stack = [amplitude_dict]
-
-    while dict_element_stack:
-        element = dict_element_stack.pop()
-
-        if (
-            "Component" in element
-            and element["Component"] == coefficient_amplitude_name
-        ):
-            # we found what we are looking for, extract the prefactor
-            if "PreFactor" in element:
-                return element["PreFactor"]["Real"]
-            return 1.0
-
-        if "Intensity" in element:
-            sub_intensity = element["Intensity"]
-            if isinstance(sub_intensity, dict):
-                dict_element_stack.append(sub_intensity)
-            elif isinstance(sub_intensity, list):
-                for sub_intensity_dict in sub_intensity:
-                    dict_element_stack.append(sub_intensity_dict)
-        elif "Amplitude" in element:
-            sub_amplitude = element["Amplitude"]
-            if isinstance(sub_amplitude, dict):
-                dict_element_stack.append(sub_amplitude)
-            elif isinstance(sub_amplitude, list):
-                for sub_amplitude_dict in sub_amplitude:
-                    dict_element_stack.append(sub_amplitude_dict)
-
-    return None
+    prefactors = amplitude_model.parameters.filter(
+        lambda p: p.name.startswith("PreFactor")
+    )
+    if len(prefactors) > 0:
+        product = 1.0
+        for factor in prefactors.values():
+            product *= factor.value
+        assert relative_parity_prefactor == product
```

Comment:
I can't comprehend if this test is doing what it should be. So two very specific Coefficient amplitudes are supposed to have a this prefactor relation with each other. This code seems to check something else, as it just takes all defined prefactors and just multiplies up those values... Maybe I misunderstood something

Reply:
Actually I couldn't figure out what the existing test was doing. But I did notice that only one of the two cases actually had a PreFactor, so the other was just testing 1 = 1*1 because:
https://github.com/ComPWA/expertsystem/blob/e6471989258071ab2480fbf6b235c5a03bab05fb/tests/unit/amplitude/test_parity_prefactor.py#L117-L119

breaks the search loop upon finding the first `"Component"`.

However, note that e58141f2 does not break the test even though it should. So something is also going wrong in this test.

Reply:
Removed `register_parameter` 84eebf79

Reply:
Old test behaviour is back edeeb6d5